executables/collector/src/main/kotlin/de/cyface/collector/handler/MeasurementHandler.kt

```diff
@@ -0,0 +1,573 @@
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with the Cyface Data Collector. If not, see <http://www.gnu.org/licenses/>.
+ */
+package de.cyface.collector.handler
+
+import de.cyface.api.Authorizer
+import de.cyface.api.PauseAndResumeBeforeBodyParsing
+import de.cyface.api.model.User
+import de.cyface.collector.handler.exception.*
+import de.cyface.collector.model.Measurement
+import de.cyface.collector.verticle.Config
+import de.cyface.model.RequestMetaData
+import io.vertx.core.MultiMap
+import io.vertx.core.file.AsyncFile
+import io.vertx.core.file.FileProps
+import io.vertx.core.file.OpenOptions
+import io.vertx.core.http.HttpServerRequest
+import io.vertx.core.json.JsonObject
+import io.vertx.ext.auth.mongo.MongoAuthentication
+import io.vertx.ext.mongo.GridFsUploadOptions
+import io.vertx.ext.mongo.MongoClient
+import io.vertx.ext.mongo.MongoGridFsClient
+import io.vertx.ext.web.RoutingContext
+import io.vertx.ext.web.Session
+import org.apache.commons.lang3.Validate
+import org.slf4j.LoggerFactory
+import java.io.File
+import java.nio.file.Path
+import java.nio.file.Paths
+import java.util.*
+import java.util.stream.Collectors
+
+/**
+ * A handler for receiving HTTP PUT requests on the "measurements" end point.
+ * This end point is the core of this application and responsible for receiving
+ * new measurements from any measurement device and forwarding those
+ * measurements for persistent storage.
+ *
+ * @author Armin Schnabel
+ * @version 1.0.0
+ * @since 6.0.0
+ */
+class MeasurementHandler(
+    mongoClient: MongoClient, authProvider: MongoAuthentication?,
+    payloadLimit: Long
+) : Authorizer(authProvider, mongoClient, PauseAndResumeBeforeBodyParsing()) {
+    /**
+     * Vertx `MongoClient` used to access the database to write the received data to.
+     */
+    private val mongoClient: MongoClient
+
+    /**
+     * The maximum number of `Byte`s which may be uploaded.
+     */
+    private val payloadLimit: Long
+
+    /**
+     * Creates a fully initialized instance of this class.
+     *
+     * @param config the configuration setting required to start the HTTP server
+     */
+    constructor(config: Config) : this(config.database, config.authProvider, config.measurementLimit) {}
+
+    /**
+     * Creates a fully initialized instance of this class.
+     *
+     * @param mongoClient The client to use to access the Mongo database.
+     * @param authProvider An auth provider used by this server to authenticate against the Mongo user database
+     * @param payloadLimit The maximum number of `Byte`s which may be uploaded.
+     */
+    init {
+        Validate.notNull(mongoClient)
+        Validate.isTrue(payloadLimit > 0)
+        this.mongoClient = mongoClient
+        this.payloadLimit = payloadLimit
+    }
+
+    override fun handleAuthorizedRequest(
+        ctx: RoutingContext, users: Set<User>,
+        header: MultiMap
```

Comment:
weird format: ether each parameter in a new line or all 3 in one line, as they are short enough